docs/website/blog/deploy-and-connect-to-microservice-using-helm-and-exec.md

```diff
@@ -0,0 +1,815 @@
+  apply:
+    component: outerloop-serviceaccount
+- id: k8s-role-for-helm
+  apply:
+    component: outerloop-role
+- id: k8s-rolebinding-for-helm
+  apply:
+    component: outerloop-rolebinding
+
+# This is the main "composite" command that will run all below commands
+- id: deploy
+  composite:
+    commands:
+      - k8s-sa
+      - k8s-role
+      - k8s-rolebinding
+      - deploy-db
+      - build-image
+      - k8s-deployment
+      - k8s-service
+      - k8s-url
+    group:
+      isDefault: true
+      kind: deploy
+# Below are the commands and their respective components that they are "linked" to deploy
+- id: build-image
+  apply:
+    component: outerloop-build
+- id: k8s-deployment
+  apply:
+    component: outerloop-deployment
+- id: k8s-service
+  apply:
+    component: outerloop-service
+- id: k8s-url
+  apply:
+    component: outerloop-url
+# highlight-end
+components:
+- container:
+    args:
+      - tail
+      - -f
+      - /dev/null
+    endpoints:
+      - name: http-go
+        targetPort: 8080
+    image: registry.access.redhat.com/ubi9/go-toolset:latest
+    memoryLimit: 1024Mi
+    mountSources: true
+  name: runtime
+# highlight-start
+- name: deploy-db
+  container:
+    image: quay.io/tkral/devbox-demo-devbox
+  attributes:
+    pod-overrides:
+      spec:
+        serviceAccountName: my-go-app
+- name: outerloop-serviceaccount
+  kubernetes:
+    inlined: |
+      apiVersion: v1
+      kind: ServiceAccount
+      metadata:
+        name: {{RESOURCE_NAME}}
+- name: outerloop-role
+  kubernetes:
+    inlined: |
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: Role
+      metadata:
+        name: {{RESOURCE_NAME}}
+      rules:
+      - apiGroups:
+        - '*'
+        resources:
+        - '*'
+        verbs:
+        - '*'
+- name: outerloop-rolebinding
+  kubernetes:
+    inlined: |
+      apiVersion: rbac.authorization.k8s.io/v1
+      kind: RoleBinding
+      metadata:
+        name: {{RESOURCE_NAME}}
+      roleRef:
+        apiGroup: rbac.authorization.k8s.io
+        kind: Role
+        name: {{RESOURCE_NAME}}
+      subjects:
+      - kind: ServiceAccount
+        name: {{RESOURCE_NAME}}
+# This will build the container image before deployment
+- name: outerloop-build
+  image:
+    dockerfile:
+      buildContext: ${PROJECT_SOURCE}
+      rootRequired: false
+      uri: ./Dockerfile
+    imageName: "{{CONTAINER_IMAGE}}"
+# This will create a Deployment in order to run your container image across
+# the cluster.
+- name: outerloop-deployment
+  kubernetes:
+    inlined: |
+      kind: Deployment
+      apiVersion: apps/v1
+      metadata:
+        name: {{RESOURCE_NAME}}
+      spec:
+        replicas: 1
+        selector:
+          matchLabels:
+            app: {{RESOURCE_NAME}}
+        template:
+          metadata:
+            labels:
+              app: {{RESOURCE_NAME}}
+          spec:
+            containers:
+              - name: {{RESOURCE_NAME}}
+ image: {{CONTAINER_IMAGE}}
```

Comment:
Looks like this (and all the container spec definition below) is not indented correctly..